sqllogictest/src/runner.rs

```diff
@@ -875,6 +877,35 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
         }
     }
 
+    /// Run a single record with retry.
+    pub async fn run_async_with_retry(
```

Comment:
It seems confusing to introduce a new method specifically to make `retry` work. For example, `run_parallel_async` does not hit this path.

I suggest moving the retry logic into `apply_record` instead, given that it's a per-record modifier.

Reply:
I feel that it's a bit confusing to do retry in `sqllogictest --override`, so I did it outside of `apply_record`.


> For example, run_parallel_async does not hit this path.

How about rename `run_async_with_retry` to `run_async` to ensure all path are covered?